adding dotnet format docs (#25082)
dotnet/docs: docs/core/tools/dotnet-format.md

---
title: dotnet format command

> description: The dotnet format command formats code to match `editorconfig` settings for the current directory.

Reviewer: Markdown formatting (backticks in this case) doesn't work in the YAML metadata block.  Also I'd follow the precedent of the VS docs which tend to prefer EditorConfig with no formatting.  Occasionally they refer to the *.editorconfig* file in Markdown, i.e., as a file name with [italic style rather than code style](https://review.docs.microsoft.com/en-us/help/contribute/text-formatting-guidelines?branch=master#italics). 

```suggestion
description: The dotnet format command formats code to match EditorConfig settings for the current directory.
```

ms.date: 07/12/2021
---
# dotnet format

**This article applies to:** ✔️ .NET 6.x SDK and later versions

## Name

`dotnet format` - Formats code to match `editorconfig` settings.

## Synopsis

```dotnetcli
dotnet format [options] [<PROJECT | SOLUTION>]

dotnet clean -h|--help
```

## Description

> `dotnet format` is a code formatter that applies style preferences to a project or solution. Preferences will be read from an .`editorconfig` file, if present, otherwise a default set of preferences will be used. For more information on `editorconfig` files see the documentation [here](https://docs.microsoft.com/dotnet/fundamentals/code-analysis/configuration-files#`editorconfig`).

Reviewer 2: "here" should be avoided as link text. Also it may be better to use relative link.

## Arguments

`PROJECT | SOLUTION`

The MSBuild project or solution to run code formatting on. If a project or solution file is not specified, MSBuild searches the current working directory for a file that has a file extension that ends in *proj* or *sln*, and uses that file.

## Options

None of the options below are required for the `dotnet format` command to succeed but can be used to further customize what is formatted and by which rules.

* **`--diagnostics <DIAGNOSTICS>`**

  A space separated list of diagnostic ids to use as a filter when fixing code style or 3rd party issues. Default value is whichever ids are listed in the `editorconfig` file. For a list of built-in analyzer rule ids that you can specify see the [list of ids for code-analysis quality rules](https://docs.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/#index-of-rules).

* **`--severity`**

>   The severity of diagnostics to fix. Allowed values are `info`, `warn`, and `error`. The default value is `warn`

Author: It include all "higher" levels above it so `info` implies `warn` and `error`

* **`--no-restore`**

  Doesn't execute an implicit restore before formatting. Default is to do implicit restore.

* **`--verify-no-changes`**

  Verifies that no formatting changes would be performed. Terminates with a non zero exit code if any files would have been formatted.

* **`--include <INCLUDE>`**

  A space-separated list of relative file or folder paths to include in formatting. All files in the solution or project are formatted if empty.

* **`--exclude <EXCLUDE>`**

  A space-separated list of relative file or folder paths to exclude from formatting.  The default is none.

* **`--include-generated`**

  Formats files generated by the SDK.

* **`-v|--verbosity <LEVEL>`**

  Sets the verbosity level. Allowed values are `q[uiet]`, `m[inimal]`, `n[ormal]`, `d[etailed]`, and `diag[nostic]`. Default value is `m[inimal]`.

* **`--binarylog <BINARY-LOG-PATH>`**

  Logs all project or solution load information to a binary log file.

* **`--report <REPORT-PATH>`**

  Produces a JSON report in the directory specified by `<REPORT_PATH>`.

* **`-h|--help`**

  Shows help and usage information

## Subcommands

### Whitespace

> `dotnet format whitespace` - Formats code to match `editorconfig` settings for whitespace.

Reviewer 2: What's the difference between this command and `dotnet format --fix-whitespace`? Might worth adding a note about that.

#### Description

> The `dotnet format whitespace` subcommand will only run formatting rules associated with whitespace formatting. For a complete list of possible formatting options that you can specify in your *.editorconfig* file see the [whitespace formatting documentation](/visualstudio/ide/reference/options-text-editor-csharp-formatting).

Reviewer 2: ```suggestion
The `dotnet format whitespace` subcommand will only run formatting rules associated with whitespace formatting. For a complete list of possible formatting options that you can specify in your *.editorconfig* file, see the [whitespace formatting documentation](/visualstudio/ide/reference/options-text-editor-csharp-formatting).
```

Reviewer 2: I personally don't find the linked documentation adding anything. It shows a screenshot from the Visual Studio options, rather than actual `.editorconfig` entries for whitespace formatting.

@gewarren, is EditorConfig formatting options documented somewhere?

Reviewer 3: I agree that's not a useful page to link to. What about this one? https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/formatting-rules#c-formatting-rules

### Style

`dotnet format style` - Formats code to match EditorConfig settings for code style.

#### Description

> The `dotnet format style` subcommand will only run formatting rule associated with code style formatting. For a complete list of possible formatting options that you can specify in your `editorconfig` file see the [code style documentation](../../fundamentals/code-analysis/style-rules/language-rules.md).

Reviewer 2: Unclear to me why the link says "code style" but links to a specific category of code style rules? for example, "[unnecessary code](https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/unnecessary-code-rules)" rules?

Author: linked to index

#### Options

* **`--severity`**

  The minimum severity of diagnostics to fix. Allowed values are `info`, `warn`, and `error`. The default value is `warn`

### Analyzers

`dotnet format analyzers` - Formats code to match `editorconfig` settings for analyzers.

#### Description

The `dotnet format analyzers` subcommand will only run formatting rule associated with analyzers. For a list of possible analyzer rules that you can specify in your `editorconfig` file see the  [list of ids for code-analysis quality rules](../../fundamentals/code-analysis/quality-rules/index.md).

> ##### Options

Reviewer 3: ```suggestion
#### Options
```

* **`--diagnostics <DIAGNOSTICS>`**

>   A space separated list of diagnostic ids to use as a filter when fixing code quality or 3rd party issues. Default value is whichever ids are listed in the `editorconfig` file. For a list of built-in analyzer rule ids that you can specify see the list of ids see the [documentation of code-analysis quality rules](../../fundamentals/code-analysis/quality-rules/index.md).

Reviewer 3: ```suggestion
  A space-separated list of diagnostic IDs to use as a filter when fixing code quality or 3rd party issues. The default value is whichever IDs are listed in the `editorconfig` file. For a list of built-in analyzer rule IDs that you can specify, see [code-analysis quality rules](../../fundamentals/code-analysis/quality-rules/index.md).
```

>   A space separated list of diagnostic ids to use as a filter when fixing code style or 3rd party issues. Default value is whichever ids are listed in the `editorconfig` file. For a list of built-in analyzer rule ids that you can specify see the list of ids see the [documentation of code-analysis quality rules](../../fundamentals/code-analysis/quality-rules/index.md).

Reviewer 2: Did this mean to say "code quality" instead of "code style"?

```suggestion
  A space separated list of diagnostic ids to use as a filter when fixing code quality or 3rd party issues. Default value is whichever ids are listed in the `editorconfig` file. For a list of built-in analyzer rule ids that you can specify see the list of ids see the [documentation of code-analysis quality rules](../../fundamentals/code-analysis/quality-rules/index.md).
```

* **`--severity`**

  The minimum severity of diagnostics to fix. Allowed values are `info`, `warn`, and `error`. The default value is `warn`

## Examples

* Format all code in the solution:

  ```dotnetcli
  dotnet format ./solution.sln
  ```

> * Clean up all code in the application project:

Reviewer 3: ```suggestion
* Format all code in the application project:
```

> * Clean up all code in the solution:

Reviewer: "Clean up" is left over from when the command name was cleanup.

```suggestion
* Format all code in the solution:
```

  ```dotnetcli
  dotnet format ./src/application.csproj
  ```

* Verify that all code is correctly formatted:

  ```dotnetcli
  dotnet format --verify-no-changes
  ```
  

> * Clean up all code in the *src* and *tests* directory but not in *src/submodule-a*:

Reviewer 3: ```suggestion
* Format all code in the *src* and *tests* directory but not in *src/submodule-a*:
```

> * Clean up all code in the `src` and `tests` directory but not in `src/submodule-a`:

Reviewer: Italics for file and folder names

```suggestion
* Clean up all code in the *src* and *tests* directory but not in *src/submodule-a*:
```
